src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule712paragraphs/InputIncorrectJavadocParagraph.java

```diff
@@ -143,9 +139,12 @@ class InnerPrecedingPtag {
      *  </ul>
      *</p>
      */
+    // 2 violations 6 lines above:
+    //  '<p> tag should be placed immediately before the first word'
+    //  '<p> tag should be preceded with an empty line.'
     public void foo() {}
```

Comment:
this is actually false positives.
Google rule for P should be applicable only for P tags  that are not nested in other HTML tags.
in other words they are on root level of javadoc.

`<p> tag should be preceded with an empty line.` this is not html rule, html does not care about empty lines.
this rule is for humans who read javadoc as source, humans more easy to parse javadoc source in this case. I doubt empty line will do a difference in resulted rendered html from javadoc.

`'<p> tag should be placed immediately before the first word'` this is also not html related rule, it is just style of Google.

When first P is open, all nested in it, is HTML, and will have content to render well in browser, there could be any layout, we should not deal with it, it is very complicated and I doubt anybody care about this.

Lets keep our logic simple, if we detect P tag, and it is nested in other tag, we skip it.

Reply:
should we check if it precedes a block-tag or we should skip that too?

Reply:
if P nested, no violations at all, no any violations.

Reply:
please document this, users should know abou this.